docs/spec/sources.md

```diff
@@ -0,0 +1,338 @@
+  - name: Age
+    type: date
+    JSONPath: .metadata.creationTimestamp
+```
+
+### Source Validation
+
+Sources MUST implement conditions with a `Ready` condition for long lived
+sources, and `Succeeded` for batch style sources.
+
+Sources MUST propagate the `sinkUri` to their status to signal to the cluster
+where their events are being sent.
+
+Knative has standardized on the following minimum OpenAPI definition of `status`
+for Source CRDs:
+
+```yaml
+validation:
+  openAPIV3Schema:
+    properties:
+      status:
+        type: object
+        properties:
+          conditions:
+            type: array
+            items:
+              type: object
+              properties:
+                lastTransitionTime:
+                  type: string
+                message:
+                  type: string
+                reason:
+                  type: string
+                severity:
+                  type: string
+                status:
+                  type: string
+                type:
+                  type: string
+              required:
+                - type
+                - status
+          sinkUri:
+            type: string
+```
+
+Please see
+[SourceStatus](https://godoc.org/github.com/knative/pkg/apis/duck/v1#SourceStatus)
+and [Condition](https://godoc.org/github.com/knative/pkg/apis/#Condition) for
+more details.
+
+Sources SHOULD provide OpenAPI validation for the `spec` field. At minimum
+sources SHOULD have the following,
+
+```yaml
+validation:
+  openAPIV3Schema:
+    properties:
+      spec:
+        type: object
+        required:
+          - sink
+        properties:
+          sink:
+            type: object
+            description:
+              "Reference to an object that will resolve to a domain name to use
+              as the sink."
+          ceOverrides:
+            type: object
+            description:
+              "Defines overrides to control modifications of the event sent to
+              the sink."
+            properties:
+              extensions:
+                type: object
+                description:
+                  "Extensions specify what attribute are added or overridden on
+                  the outbound event. Each `Extensions` key-value pair are set
+                  on the event as an attribute extension independently."
+```
+
+### Event Type Registry
+
+The Event Type Registry is a feature that allows for software to find all CRDs
+labeled as a `Source` ducktype, and extract the defined
+[EventType](https://godoc.org/knative.dev/eventing/pkg/apis/eventing/v1alpha1#EventType)s
+that could be emitted from a source instance.
+
+To be included in the event type registry a fake `registry` component is defined
+in the `openAPIV3Schema` section inside the CRD. It is not our intention to have
+cluster operators create resources with this filled out, it is to signal to
+other software some metadata about the runtime/dataplane characteristics of this
+type of source.
```

Comment:
As laid out in https://github.com/knative/eventing/issues/1550#issuecomment-519791678 I still think that this approach is very hackish, and e.g. "misusing" the `pattern` schema pattern field fon non-pattern like (e.g. non-ECMA 262 regular expressions).

IMO everything specified in a schema should be part of an instance (i.e. being an instance variable which can be set by a user, not used as a static enum-type without meaning).

Reply:
Good points, I had missed this discussion. I agree that it needs more discussion, I commented out the section but did not deleted it yet. Let's talk more. Thanks for bringing it up.

Reply:
As proposed in https://github.com/knative/eventing/issues/1550#issuecomment-532722794 and  https://github.com/knative/eventing/issues/1550#issuecomment-532723930 moving the registry meta-data to annotations would make much sense. Would this be something which should be changed here or should the change be postponed to a later version of this spec ?

Reply:
let's address the registry as a followup so we can talk about it more as an independent feature.

Reply:
@rhuss should we move this forward? Or dont feel this is "ready" ?

Reply:
It seems like any mention of a registry could be a follow-on PR.

Maybe remove this commented section from the PR and follow up in a separate PR addressing #1550?

Reply:
Removed and added a comment pointing to #1550